### ISSUE TITLE
Keeping timers relative to start of tunnel initiation is confusing

### ISSUE THREAD

**Issue by agrover** [cloudflare/boringtun#309]
Keeping timers this way means that for example `update_session_timers` takes a `time_now` param that's a `Duration` instead of an `Instant`. All entries in `Tunn::timers` are `Duration`s when many of them are "the time that something happened" and idiomatically I think these would be better expressed as `Instant`s.

**Comment by Noah-Kennedy** [comment]
Agreed.

**Comment by crajcan** [comment]
@agrover @Noah-Kennedy Anyone available to review?

**Comment by thomaseizinger** [comment]
Friendly ping! This would be a nice addition :)

**Comment by thomaseizinger** [comment]
> > Friendly ping! This would be a nice addition :)
> 
> @thomaseizinger does my PR accomplish what you're looking for? If so I'll rebase it.

Yes! I am hoping that we can also move this to be entirely SANS-IO so we have control over the what "now" is :)

See https://github.com/cloudflare/boringtun/issues/391.

**Comment by bountis** [comment]
Hey, I see that this has been stale for a while, is this still relevant? I would love to work on this if so!
